services/auth/source/ldap/source_sync.go

```diff
@@ -41,9 +39,15 @@ func (source *Source) Sync(ctx context.Context, updateExisting bool) error {
 	default:
 	}
 
-	sort.Slice(users, func(i, j int) bool {
-		return users[i].LowerName < users[j].LowerName
-	})
+	usernameUsers := make(map[string]*user_model.User, len(users))
+	mailUsers := make(map[string]*user_model.User, len(users))
+	existingUsers := make(map[int64]struct{})
+
+	for i := range users {
+		u := users[i]
+		usernameUsers[u.LowerName] = u
+		mailUsers[u.Email] = u
```

Comment:
Are the `Email` and `Mail` case-sensitive or case-insensitive?

Reply:
The `Mail` from LDAP might contain uppercase letters so I think we should convert them to lowercase before comparing.
Done in 003ea013cc019eed323473f604f8da4050f44442